doc/sphinx/proofs/writing-proofs/equality.rst

```diff
@@ -26,11 +26,22 @@ There are multiple notions of :gdef:`equality` in Coq:
 
 - :gdef:`Definitional equality <definitional equality>` is equality based on the
   :ref:`conversion rules <Conversion-rules>`, which Coq can determine automatically.
-  When two terms are definitionally equal, Coq knows it can
+  Two terms are definitionally equal when they reduce to syntactically identical terms
+  using the conversion rules.  When two terms are definitionally equal, Coq knows it can
   replace one with the other, such as with :tacn:`change` `X with Y`, among many
   other advantages.  ":term:`Convertible <convertible>`" is another way of saying that
   two terms are definitionally equal.
 
+  Among other reductions, the conversion rules can do computation to simplify
+  expressions.  The behavior depends on the type associated with the
+  operator and operands.  For example, for `nat` and `Z`, `1 + 2` is reduced to `3`.
```

Comment:
I see the point you are trying to express here, but I believe it is not perfectly accurate as currently worded. The behavior does not depend on the type but on the definition of the operator. The fact that `1 + 2` behaves differently in `nat` and `R` is because the operator is different, despite using the same notation (in a different scope, which is left implicit here).

Overall, I think this paragraph is valuable for people with a mathematical background, which might be surprised by the behavior described here. So let's try to amend it to make it 100% accurate, while not changing its substance too much.

I suggest changing this sentence with something like: "When trying to reduce an operation containing operators, Coq looks at the definition of the operator, which may differ depending on which notation scope is open, and thus may change the behavior of the reduction."

Then, the part "For example, for `nat` and `Z`", should be adjusted to "For example, for the standard library definitions of `+` for `nat` and `Z`," and the rest of the paragraph can be minimally adjusted to avoid talking about "types" and to talk about operator definitions instead.

Reply:
Updated (in 2 pushes), WDYT?